src/components/submission/list.vue

```diff
@@ -305,14 +311,25 @@ export default {
   min-height: 375px;
 }
 
-#submission-filters + #submission-field-dropdown { margin-left: 15px }
-#submission-filters + #submission-list-refresh-button { margin-left: 10px; }
-#submission-field-dropdown + #submission-list-refresh-button {
+#submission-list-actions {
+  align-items: baseline;
+  display: flex;
+  flex-wrap: wrap-reverse;
+}
+#submission-field-dropdown {
   margin-left: 15px;
+  margin-right: 5px;
+}
+#submission-list-refresh-button {
+  margin-left: 10px;
+  margin-right: 5px;
 }
 #submission-download-dropdown {
-  float: right;
-  top: 3px;
+  margin-bottom: 10px;
+  margin-left: auto;
+
+  // Needed to work with align-items above.
+  .btn { float: none; }
```

Comment:
Bootstrap floats `.btn-group > .btn` to the left, but I think it's not needed for this dropdown (or maybe for any of our uses of `.btn-group`?).